### PR TITLE
layersData closes #6564

### LINKED ISSUE #6564: Make layer data available on editor first load
Since layers data isn't available when the page is loaded we need to do an extra round-trip to the API to get the associated layers of the visualization; would it be possible to have this information available...

a) as part of the [visualizationData](https://github.com/CartoDB/cartodb/blob/master/app/views/carto/editor/visualizations/show.html.erb#L11)?
b) as a new layerData variable?

Which output is needed, @viddo? (detailed, vizjson...). It'd be great having a needed output example if you have one.

### PR DESCRIPTION
**Pull request by juanignaciosl**
@viddo, would you check if this is what you need, please?

screen shot 2016-02-12 at 12 17 17

### CODE REVIEW

**Comment by viddo** [comment]
:ok_hand: 
updated the entry point to use the new provided data :kissing_heart:

:shipit: whenever the CI is :green_heart:

**Comment by juanignaciosl** [comment]
@gfiorav CR this, please.

File: app/controllers/carto/editor/visualizations_controller.rb [hunk 1]
```diff
@@ -15,6 +18,7 @@ class VisualizationsController < EditorController
 
       def show
         @visualization_data = Carto::Api::VisualizationPresenter.new(@visualization, current_viewer, self).to_poro
+        @layers_data = @visualization.layers.map { |l| Carto::Api::LayerPresenter.new(l).to_poro }
```
Comment (gfiorav) [inline app/controllers/carto/editor/visualizations_controller.rb]: Curious: The layer presenter shows no info about widgets right?

**Comment by gfiorav** [comment]
Fine with me :+1:

File: app/controllers/carto/editor/visualizations_controller.rb (hunk 1 above)
Comment (juanignaciosl) [inline app/controllers/carto/editor/visualizations_controller.rb]: Nop. Frontenders decided to keep widgets outside layers from a presentation point of view. In the vizjson they're rendered as a first level attribute indeed.

Comment (gfiorav) [inline app/controllers/carto/editor/visualizations_controller.rb]: ok, makes sense!

Thanks for explaining!

**Pull request merged**